doc/stdenv/stdenv.chapter.md

```diff
@@ -116,6 +116,65 @@ On Linux, `stdenv` also includes the `patchelf` utility.
+- libraries used by compilers, for example `zlib`,
+- interpreters needed by [`patchShebangs`](#patch-shebangs.sh) for scripts which are installed, which can be the case for e.g. `perl`
+
+Note that these criteria are not incompatible: for example software using Wayland usually needs the `wayland` library at runtime (so `wayland` should be added to `buildInputs`) but also executes the `wayland-scanner` program as part of the build to generate code (so `wayland` should also be added to `nativeBuildInputs`).
+
+Dependencies needed only to run tests are similarly classified between native (executed during build) and non-native (executed at runtime):
+- `nativeCheckInputs` for test tools needed on `$PATH` (like `ctest`) and [setup hooks](#ssec-setup-hooks) (for example [`pytestCheckHook`](#python))
+- `checkInputs` for libraries linked into test executables (for example the `qcheck` OCaml package)
+
+These dependencies are only injected when `doCheck` is set to `true`.
+
+Consider for example this simplified derivation for `solo5`, a sandboxing tool:
+```nix
+stdenv.mkDerivation rec {
+  pname = "solo5";
+  version = "0.7.5";
+
+  src = fetchurl {
+    url = "https://github.com/Solo5/solo5/releases/download/v${version}/solo5-v${version}.tar.gz";
+    sha256 = "sha256-viwrS9lnaU8sTGuzK/+L/PlMM/xRRtgVuK5pixVeDEw=";
+  };
+
+  nativeBuildInputs = [ makeWrapper pkg-config ];
+  buildInputs = [ libseccomp ];
+
+  postInstall = ''
+    substituteInPlace $out/bin/solo5-virtio-mkimage \
+      --replace "/usr/lib/syslinux" "${syslinux}/share/syslinux" \
+      --replace "/usr/share/syslinux" "${syslinux}/share/syslinux" \
+      --replace "cp " "cp --no-preserve=mode "
+
+    wrapProgram $out/bin/solo5-virtio-mkimage \
+      --prefix PATH : ${lib.makeBinPath [ dosfstools mtools parted syslinux ]}
+  '';
+
+  doCheck = true;
+  nativeCheckInputs = [ util-linux qemu ];
+  checkPhase = '' [elided] '';
+}
+```
+
+`makeWrapper` is a setup hook, i.e. a shell script sourced by the generic builder of `stdenv`. It is thus executed during build and must be added to `nativeBuildInputs`. `pkg-config` is a build tool which the configure script of `solo5` expects to be on `$PATH` during build: therefore, it must be added to `nativeBuildInputs`. `libseccomp` is a library referenced by `$out/bin/solo5-elftool`. As it is used at runtime, it must be added to `buildInputs`. Tests need `qemu` and `getopt` (from `util-linux`) on `$PATH`, these must be added to `nativeCheckInputs`.
+Finally, notice that some dependencies are injected directly in the shell code of phases: `syslinux`, `dosfstools`, `mtools`, and `parted`. This is perfectly fine. Nix will detect that they are present in the output `$out` and will mark them as runtime dependencies even if they are not present in `buildInputs`.
+
+For more complex cases, like libraries linked into an executable which is then executed as part of the build system, read on the next section.
```

Comment:
Sections may be reordered, and implying an order in the prose is prone to dangling pointers. Please use an explicit link, IIRC there is a way to expand headings automatically. Since I don't know the manual by heart, honestly I'm not even sure what you refer to exactly, because the next section is titled Reference and that doesn't tell me how it will answer the question.

Reply:
`read [](#ssec-stdenv-dependencies-reference)` should work.